test/SILGen/lazy_property_with_observers.swift

```diff
@@ -0,0 +1,58 @@
+// RUN: %target-swift-emit-silgen %s | %FileCheck %s
+
+class Foo {
+  lazy var bar: Int = 0 {
+    didSet { print(oldValue) }
+    willSet { print(newValue) }
+  }
+
+  lazy var baz: Int = 0 {
+    didSet { }
+    willSet { print(newValue) }
+  }
+}
+
+// Foo.bar.setter
+
+// CHECK-LABEL: sil hidden [ossa] @$s28lazy_property_with_observers3FooC3barSivs : $@convention(method) (Int, @guaranteed Foo) -> () {
+// CHECK: bb0([[VALUE:%.*]] : $Int, [[FOO:%.*]] : @guaranteed $Foo):
+// CHECK-NEXT:  debug_value [[VALUE]] : $Int, let, name "value", argno 1 // id: %2
+// CHECK-NEXT:  debug_value [[FOO]] : $Foo, let, name "self", argno 2 // id: %3
+// CHECK-NEXT:  [[GETTER:%.*]] = class_method [[FOO]] : $Foo, #Foo.bar!getter : (Foo) -> () -> Int, $@convention(method) (@guaranteed Foo) -> Int
```

Comment:
If a subclass overrides the property, this calls the overridden getter. Is that expected? With a plain old stored property, the oldValue in the observer goes direct to storage.

Did we ever do that thing where if you don't touch the oldValue from didSet, we avoid loading it entirely?

Reply:
Hmm, the storage would be `nil` initially hence why the getter is called.

And yeah, I have a test case for when we have a simple didSet which doesn't fetch the oldValue. In that case, we don't call the getter :)

Reply:
@slavapestov Does that answer your question?